app/controllers/group_offers_controller.rb

```diff
@@ -0,0 +1,56 @@
+class GroupOffersController < ApplicationController
+  before_action :set_group_offer, only: [:show, :edit, :update, :destroy]
+
+  def index
+    authorize GroupOffer
+    @group_offers = policy_scope(GroupOffer)
+  end
+
+  def show; end
+
+  def new
+    @group_offer = GroupOffer.new
+    authorize @group_offer
+  end
+
+  def edit; end
+
+  def create
+    @group_offer = GroupOffer.new(group_offer_params)
+    @group_offer.department = current_user.department.first if current_user.manages_department?
```

Comment:
and what happens to the superadmin adding Group offers to different departments?
Can superadmins do that now?

Reply:
The only users that have departments are the department managers. So this line affects only them.

Reply:
Ah, now I realize. The association is optional. It makes sense then.